airflow/contrib/operators/app_engine_operator.py

```diff
@@ -40,15 +43,18 @@ def schedule_job(self):
     def poll_status_files(self):
         success_file_name = '%s/succeeded' % self.job_id
         fail_file_name = '%s/failed' % self.job_id
+        start_time = datetime.utcnow()
         i = 0
-        while True:
-            time.sleep(5 * 2**i)
+        while (datetime.utcnow() - start_time).total_seconds() < 3600:
```

Comment:
I decided to back-port the max polling interval and max polling time to the existing AppEngineOperator. Prevents very long polls and/or timeouts.